hiero-enterprise-base/src/main/java/com/openelements/hiero/base/implementation/AccountClientImpl.java

```diff
@@ -24,10 +24,23 @@ public AccountClientImpl(@NonNull final ProtocolLayerClient client) {
     @NonNull
     @Override
     public Account createAccount(@NonNull Hbar initialBalance) throws HieroException {
-        final AccountCreateRequest request = AccountCreateRequest.of(initialBalance);
-        final AccountCreateResult result = client.executeAccountCreateTransaction(request);
-        return result.newAccount();
+        if (initialBalance == null) {
+            throw new NullPointerException("initialBalance must not be null");
+        }
+        
+        if (initialBalance.toTinybars() < 0) {
+            throw new HieroException("Invalid initial balance: must be non-negative");
+        }
+        
+        try {
+            final AccountCreateRequest request = AccountCreateRequest.of(initialBalance);
+            final AccountCreateResult result = client.executeAccountCreateTransaction(request);
+            return result.newAccount();
+        } catch (IllegalArgumentException e) {
+            throw new HieroException("Invalid initial balance: " + e.getMessage(), e);
```

Comment:
@Lemeri123 I do not know for 100% if that is correct. I assume an IllegalArgumentException can have m ultiple causes and must not always be that that the balance is invalid. It would be better to have a more generic message like:
```suggestion
            throw new HieroException("Error while creating Account", e);
```